Code/Tools/AssetProcessor/native/utilities/SimpleStatsCapture.cpp

```diff
@@ -0,0 +1,392 @@
+            };
+
+            // note, keys is non const, since this function also sorts it.
+            auto printStatsArray = [&](AZStd::vector<AZStd::string>& keys, int maxToPrint, const char* optionalHeader)
+            {
+                if ((dumpStatsHumanReadable)&&(optionalHeader))
+                {
+                    AZ_TracePrintf(AssetProcessor::ConsoleChannel,"Top %i %s\n", maxToPrint, optionalHeader);
+                }
+                std::sort(keys.begin(), keys.end(), sortByTimeDescending);
+                
+                for (int idx = 0; idx < maxToPrint; ++idx)
+                {
+                    if (idx < keys.size())
+                    {
+                        printStat(keys[idx].c_str(), m_stats[keys[idx]].m_cumulativeTime, m_stats[keys[idx]].m_operationCount);
+                    }
+                }
+            };
+            
+            StatsEntry& gemLoadStat = m_stats["LoadingModules"];
+            printStat("LoadingGems", gemLoadStat.m_cumulativeTime, 1);
+            // analysis-related stats
+
+            StatsEntry& totalScanTime = m_stats["AssetScanning"];
+            printStat("AssetScanning", totalScanTime.m_cumulativeTime, totalScanTime.m_operationCount);
+            StatsEntry& totalHashTime = m_stats["HashFileTotal"];
+            printStat("HashFileTotal", totalHashTime.m_cumulativeTime, totalHashTime.m_operationCount);
+            printStatsArray(allHashFiles, maxIndividualStats, "longest individual file hashes:");
+           
+            // CreateJobs stats
+            StatsEntry& totalCreateJobs = m_stats["CreateJobsTotal"];
+            if (totalCreateJobs.m_operationCount)
+            {
+                printStat("CreateJobsTotal", totalCreateJobs.m_cumulativeTime, totalCreateJobs.m_operationCount);
+                printStatsArray(allCreateJobs, maxIndividualStats, "longest individual CreateJobs");
+                printStatsArray(allCreateJobsByBuilder, maxCumulativeStats, "longest CreateJobs By builder");
+            }
+            
+            // ProcessJobs stats
+           StatsEntry& totalProcessJobs = m_stats["ProcessJobsTotal"];
+            if (totalProcessJobs.m_operationCount)
+            {
+                printStat("ProcessJobsTotal", totalProcessJobs.m_cumulativeTime, totalProcessJobs.m_operationCount);
+                printStatsArray(allProcessJobs, maxIndividualStats, "longest individual ProcessJob");
+                printStatsArray(allProcessJobsByJobKey, maxCumulativeStats, "cumulative time spent in ProcessJob by JobKey");
+                printStatsArray(allProcessJobsByPlatform, maxCumulativeStats, "cumulative time spent in ProcessJob by Platform");
+            }
+            duration costToGenerateStats =  AZStd::chrono::high_resolution_clock::now() - startTimeStamp;
+            printStat("ComputeStatsTime", costToGenerateStats, 1);
+        }
+
+        // Public interface:
+        static SimpleStatsCaptureImpl* g_instance = nullptr;
```

Comment:
Nitpick: C++ style guide recommends `Instance` instead of the Hungarian Notation `g_instance`.

Reply:
I assume you mean that you'd prefer to see
```
         static SimpleStatsCaptureImpl* Instance = nullptr; 
```

?  do we have other code that conforms to this?  I'm seeing s_instance and g_instance all over the place.

Reply:
That's corect.

Reply:
I'm going to go ahead and follow the style guide.
https://github.com/o3de/sig-core/blob/02e5323c4e0be59581d72f2249c8b13214aaa42e/governance/Coding-Standards-and-Style-Guide.md#naming

Which currently states 
Static member variables should begin with the s_ prefix, e.g. s_monotonicCounter
Global variables should begin with a g_ prefix, e.g. g_dataForEveryone
I'll go ahead and change it to g_.